cosmwasm-contracts/marketing/src/contract.rs

```diff
@@ -0,0 +1,273 @@
+    pub(crate) live_collections: Map<'static, u64, MarketingCollectionPreview>,
+    pub(crate) highlighted_collections: Map<'static, u64, MarketingCollectionPreview>,
+}
+
+#[entry_points]
+#[contract]
+#[error(ContractError)]
+impl MarketingContract {
+  pub const fn new() -> Self {
+    Self {
+      config: Item::new("config"),
+      banners: Map::new("banners"),
+      news: Map::new("news"),
+      upcoming_collections: Map::new("upcoming_collections"),
+      live_collections: Map::new("live_collections"),
+      highlighted_collections: Map::new("highlighted_collections"),
+    }
+  }
+
+  #[msg(instantiate)]
+  pub fn instantiate(&self, ctx: InstantiateCtx
+    , admin_addr: String
+  ) -> StdResult<Response> {
+    let admin_addr = ctx.deps.api.addr_validate(admin_addr.as_str())?;
+    self.config.save(ctx.deps.storage, &Config {admin_addr})?;
+    Ok(Response::default())
+  }
+
+  #[msg(exec)]
+  // Only the admin can execute it
+  pub fn update_config(
+    &self,
+    ctx: ExecCtx,
+    admin_addr: String,
+  ) -> Result<Response, ContractError> {
+    let attributes = vec![attr("action", "update_config")];
+    self.config.update(ctx.deps.storage, |mut config| {
+      // Permission check
+      if ctx.info.sender != config.admin_addr {
+        return Err(ContractError::Unauthorized);
+      }
+      // Validate and save the new admin_addr in config
+        config.admin_addr = ctx.deps.api.addr_validate(admin_addr.as_str())?;
+
+      Ok(config)
+        })?;
+
+        Ok(Response::new().add_attributes(attributes))
+      }
+
+      #[msg(exec)]
+  // Only the admin can execute it
+  pub fn update_banners(
+    &self,
+    ctx: ExecCtx,
+    banners: Vec<Banner>,
+  ) -> Result<Response, ContractError> {
+    let attributes = vec![attr("action", "update_banners")];
+    let config = self.config.load(ctx.deps.storage)?;
+    // Permission check
+    if ctx.info.sender != config.admin_addr {
+        return Err(ContractError::Unauthorized);
+    }
+    // Replace all banners with the new ones
+    self.banners.clear(ctx.deps.storage);
+    for (index, banner) in banners.into_iter().enumerate() {
+        self.banners.save(ctx.deps.storage, index as u64, &banner)?;
+    }
+
+    Ok(Response::new().add_attributes(attributes))
+  }
+
+  #[msg(exec)]
+  // Only the admin can execute it
+  pub fn update_news(
+    &self,
+    ctx: ExecCtx,
+    news: Vec<News>,
+  ) -> Result<Response, ContractError> {
+    let attributes = vec![attr("action", "update_news")];
+    let config = self.config.load(ctx.deps.storage)?;
+    // Permission check
+    if ctx.info.sender != config.admin_addr {
+      return Err(ContractError::Unauthorized);
+    }
+    // Replace all news with the new ones
+    self.news.clear(ctx.deps.storage);
+    for (index, news_item) in news.into_iter().enumerate() {
+        self.news.save(ctx.deps.storage, index as u64, &news_item)?;
+    }
+
+    Ok(Response::new().add_attributes(attributes))
```

Comment:
I think we attributes is not use in earlier code then you can write it here to make the code more readable, we dont have to traceback to see the value of attributes

Reply:
Ok I see